addons/mail/static/src/models/channel.js

```diff
@@ -0,0 +1,109 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one, many } from '@mail/model/model_field';
+import { insertAndReplace, replace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'Channel',
+    identifyingFields: ['id'],
+    recordMethods: {
+        async fetchChannelMembers() {
+            const channelData = await this.messaging.rpc({
+                model: 'mail.channel',
+                method: 'load_more_members',
+                args: [[this.id]],
+                kwargs: {
+                    known_member_ids: this.channelMembers.map(channelMember => channelMember.partner.id),
+                },
+            });
+            if (!this.exists()) {
+                return;
+            }
+            this.update(channelData);
+        },
+        /**
+         * @private
+         * @returns {Boolean|FieldCommand}
+         */
+        _computeAreAllMembersLoaded() {
+            return this.memberCount === this.channelMembers.length;
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeThread() {
+            return insertAndReplace({
+                id: this.id,
+                model: 'mail.channel',
+            });
+        },
+        _sortMembers() {
+            return [
+                ['defined-first', 'name'],
+                ['locale-asc', 'name'],
+            ];
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeOrderedOfflineMembers() {
+            return replace(this.channelMembers.filter(member => member.partner && !member.partner.isOnline));
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeOrderedOnlineMembers() {
+            return replace(this.channelMembers.filter(member => member.partner && member.partner.isOnline));
+        },
```

Comment:
Could be more efficient by setting compute the other way around.
I mean, let `ChannelMember` compute a `one` relational field, and let the `many` inverse be built automatically and efficiently.

Reply:
Done !